src/Mvc/Mvc/test/MvcServiceCollectionExtensionsTest.cs

```diff
@@ -218,6 +252,21 @@ public void AddRazorPages_Twice_DoesNotAddDuplicates()
         VerifyAllServices(services);
     }
 
+    [Fact]
+    public void AddRazorPagesWithEntryAssembly_Twice_DoesNotAddDuplicates()
+    {
+        // Arrange
+        var services = new ServiceCollection();
+        var assembly = GetEntryAssembly();
+
+        // Act
+        services.AddRazorPages(assembly);
```

Comment:
What happens when you call this with different assemblies?

Reply:
`ApplicationPart` is already configured and we will not populate the default parts from this new assembly.

Reply:
Will it throw or noop?

Reply:
Yeah, perhaps we should add a semaphore to the `IServiceCollection` and throw to ensure you can't call this repeatedly. Typically most `IServiceCollection.Add*` are idempotent, but this one wouldn't be.

Scenarios that should be disallowed:

```
services.AddRazorPages(assemblyX);
services.AddRazorPages(assemblyY);
```
```
services.AddRazorPages();
services.AddRazorPages(assembly); // and the inverse
``

Reply:
Yeah, make sense. Because right now it will be a noop. So, should we throw only when it called repeatedly using different assemblies, right?

If that is the case, we are tracking the `entryAssembly` when the parts are populated, so, it should be an easy implementation.

Reply:
I added a verification of the `entryassembly` and throwing an `ArgumentException` when they are different.